### DISCUSSION TITLE
Timelapse Export - Frigate 0.14.0

### DISCUSSION THREAD

**Discussion by xekil** [blakeblackshear/frigate#12958] [General]
Hello,

I used to do timelapse exports over a given period, I could see in the changelog that Timelapse is no longer available in the UI but the API is still available.

How to use it to do an export ?

Thanks in advance

**Comment by NickM-27** [comment]
See the docs https://docs.frigate.video/integrations/api#post-apiexportcamerastartstart-timestampendend-timestamp

It is an api request with unix timestamps, then you would just include a json body with 

```
{
"playback": "timelapse_25x"
}
```

**Reply by xekil** [reply]
thanks for the answer, but I don't really see how to use this simply, especially from my smartphone. From where should I launch the requests? Isn't it possible to add the functionality as it was the case on frigate 13? Thanks

**Reply by LukeHandle** [reply]
For the sake of a more copy paste-able example with `curl` (though still not smartphone friendly):
```
curl -H "Content-Type: application/json" http://<Frigate IP>:5000/api/export/<Camera Name>/start/$(date -d "2024-10-01 10:00" +%s)/end/$(date -d "2024-10-02 10:00" +%s) -d '{"playback": "timelapse_25x"}'
```

Replace the Frigate IP / camera name as required and this will trigger an export for the 1st 10am -> 2nd 10am.

**Reply by xekil** [reply]
Thanks for this example, I just did some tests with this command line and I encounter several problems:

**1)** The weight of the generated Timelapse is higher than the same video in normal export and with audio:

- Timelapse video x25: Duration 5 minutes = 3.43GB
- Normal video: Duration 2h = 3.17GB (with audio)

**2)** I can't delete an export during generation, in fact, the trash in the graphical interface has no effect, the "delete" button is not activated.

From memory, all this worked well before when it was possible to generate a Timelapse with the graphical interface.

It's really a shame that this function is no longer present...

**Reply by NickM-27** [reply]
> From memory, all this worked well before when it was possible to generate a Timelapse with the graphical interface.

No, you could never delete an export while it was in progress. Nothing has changed with the timelapse generation logic either

**Reply by xekil** [reply]
ok

**Comment by ItsMeRitch** [comment]
Would also really like to have the Timelapse exports functionality back in the UI

**Reply by NickM-27** [reply]
Like it explains in the FAQ, it would be helpful to understand the use case.

**Reply by ItsMeRitch** [reply]
> Like it explains in the FAQ, it would be helpful to understand the use case.

Hey! Thanks for the prompt response. 

My use case for the Timelapse's has been to condense hours/days of footage of construction work and car movement into significantly reduced file sizes and times. 

Moreover, some of my cameras produce quite large file sizes which means I can't scrub through them easily so Timelapse's help an absolute ton with that.

**Reply by NickM-27** [reply]
The first issue won't be helped in most cases because timelapses are still pretty large with the way they were implemented in 0.13 (and still in 0.14) since they are created from the full res recordings, unless you are setting manual args in ffmpeg args. In general I would expect that the new UI serves the main use case with something like the motion tab in review which provides quick scrubbing of time periods. 

It seems like what you are describing is basically just duplicating what the UI already provides but requiring the manual step of creating the file then opening it externally

**Reply by ItsMeRitch** [reply]
Hey @NickM-27 

Thanks for the reply! 

Yeah, you are correct in that I would be using the Timelapse export function to provide a way of scrubbing through footage externally. 

My main concern with not having a Timelapse function is that when I need to review footage, there's a high possibility that I'm not at home and so, not within my LAN and my internet will be restricted/poor. I was currently away on holidays there and was trying to access my Frigate history to review footage and it was practically impossible with the slow internet I was connected to. 

For me in this scenario, I was able to create a Timelapse of the footage and leave it to sit and download in the background and then I could scrub through it on my phone easily. Granted, I could do this with a realtime export but scrubbing through the realtime export was hit and miss for me in the sense that sometimes you could scrub through footage like fast forwarding through a tv programme and other times, it was like skipping between chapters of a movie. Well, that was the case with my iPhone 14 Pro. 

Timelapse solved this for me and enabled me to scrub through footage easily.

**Reply by NickM-27** [reply]
the previews that are used for scrubbing in the UI are very low in size and they are loaded gradually as the frigate backend supports partial content requests for these files. It would be good to have more information on what you are describing because at least personally, the previews load quite quickly for scrubbing purposes even on mobile with slow data

**Reply by ItsMeRitch** [reply]
Understandable, however scrubbing via the UI is great when you're wanting to find a specific time or event and you don't need good quality but for when I already have the timeframe I want to watch, watching via the UI Timeline just results in a hazy blur with an inconsistent speed. 

Even as another scenario where I've used the Timelapse, if I wanted to create a Timelapse of my dogs running around the garden for an hour or so and send it to my family, I need to export the realtime and then manually create a Timelapse. I understand that you can use the API but that is cumbersome if you're away from home and out and about. Previously, it was as simple as chucking the timeframe in and letting the server do its thing. Then, a few minutes later, I can just let it sit and download then share. 

I totally understand where you're coming from in terms of using the timeline as an alternative but it's definitely not a replacement for the Timelapse function, for me anyway. 

Can I use the timeline to watch a fast forwarded video of the dogs playing in the garden, kind of yes. Is it as good quality, no. Is it as portable and sharable, no. 

Totally understand where yourself and the team are coming from, however, this is just my two cents worth and how and why I valued the Timelapse function. If my use cases and desires of Frigate don't match the rest of the community, then I think it's fair enough to omit, but just thought I would add to another user saying they miss the functionality. :)

**Reply by NickM-27** [reply]
thanks for entertaining the conversation, it is helpful to understand the very specific and context for why certain features are important, and of course nothing is set in stone

**Comment by xekil** [comment]
In my case, I use Timelapse a lot to condense several hours of video when my children play in the garden or in the swimming pool, or as I just read, when there is work in the garden, it is a feature that I appreciate and really on frigate 0.13, having a container still with this version I hesitate to switch back to it but the databases were not compatible it is complicated.

**Reply by NickM-27** [reply]
right, same as the other user what about the UI scrubbing feature does not work for this use case?

**Comment by xekil** [comment]
I didn't really understand what features you mean, as far as I'm concerned Frigate 14 is magnificent in every way, I just regret the absence of the Timelapse feature, everything else is fabulous. I'll take advantage of it, how do I know which object was detected in the image, I can't find the frames in the detection images anymore as was the case in the Frigiel 13? Thanks

**Reply by NickM-27** [reply]
> I didn't really understand what features you mean

what you described you used time lapses for is essentially the same that the timeline scrubbing works for is my point. 

> how do I know which object was detected in the image, I can't find the frames in the detection images anymore as was the case in the Frigiel 13? Thanks

In 0.14 the option does not exist directly, though images with bounding boxes can be viewed in the Frigate+ page and in the HA integration. 

A new "more info" pane has already been implemented for 0.15 that shows this along with other information.

**Reply by xekil** [reply]
Ok thanks you for you reply

**Comment by heebo1974** [comment]
I just today switched to version 0.14, even though I had been using the 0.13 since beta version. I am really disappointed with the decision to remove the timelapse export from the UI. I used it quite a bit to make so-called fun videos. For example, watching the couple hours snow removal after a snowstorm in timelapse 'sped up' was considered fun by many.

**Reply by hawkeye217** [reply]
The reasons for removing this from the UI are explained in the FAQ (Usage question 7): https://github.com/blakeblackshear/frigate/discussions/12724

There are many free tools available for speeding up video. Adobe Express is a free web-based tool. Ffmpeg can easily be used on the command line on many machines as well:

`ffmpeg -r (desired output frame rate) -i (input file) -c:v copy -c:a copy (output file)`

**Reply by ItsMeRitch** [reply]
@hawkeye217 

The FAQ states that "Most users were using that feature to create videos to manually review each day" but I feel like @heebo1974 myself and others are just stating that we were not using it for that reason. 

I'm curious as to how you were able to gather how the Timelapse function was being used by most users? Did you carry out surveys or is there some sort of metric reporting built into frigate. 

P.S, it's always difficult to communicate tone via online messages, in no way am I trying to be argumentative or confrontational, my query comes from a place of pure curiosity and desire to learn more and contribute. 😃 

Once again, thanks for all the team's hard work with a fantastic app.

**Reply by hawkeye217** [reply]
Thanks for your kind words! We just had many users file support requests for problems with the feature, and in the conversation they would explain their use case.

**Reply by ItsMeRitch** [reply]
> Thanks for your kind words! We just had many users file support requests for problems with the feature, and in the conversation they would explain their use case. 

I see! That makes sense. Thank you for the prompt response!
